test/powershell/Modules/Microsoft.PowerShell.Management/Set-Location.Tests.ps1

```diff
@@ -53,6 +53,22 @@ Describe "Set-Location" -Tags "CI" {
         $(Get-Location).Path | Should -BeExactly $testPath.FullName
     }
 
+    It "Should not use filesystem root folder if not in filesystem provider" -Skip:(!$IsWindows) {
+        # find filesystem root folder that doesn't exist in HKCU:
+        $foundFolder = $false
+        foreach ($folder in Get-ChildItem "${env:SystemDrive}\" -Directory) {
+            if (-Not (Test-Path "HKCU:\$($folder.Name)")) {
+                $testFolder = $folder.Name
+                $foundFolder = $true
+                break
+            }
+        }
+        $foundFolder | Should -BeTrue
+        Set-Location HKCU:\
```

Comment:
Use `Push-Location HKCU:\`  in try and `Pop-Location` in finally to reset the current working directory.

Reply:
This is for `Set-Location` tests, though

Reply:
Ok. closed.